lib/datadog/statsd.rb

```diff
@@ -223,6 +223,9 @@ def initialize(
       raise ArgumentError, 'tags must be a Array<String>' unless tags.nil? or tags.is_a? Array
       @tags = (tags || []).compact.map! {|tag| escape_tag_content(tag)}
 
+      # append the entity id to tags if DD_ENTITY_ID env var is not nil
+      @tags << 'dd.internal.entity_id:' + ENV.fetch('DD_ENTITY_ID', nil) unless ENV.fetch('DD_ENTITY_ID', nil).nil?
```

Comment:
Small nit: there seems to be a `escape_tag_content` function called above, we should probably also apply it to DD_ENTITY_ID to stay consistent

Reply:
make sense, thanks